src/data/nat/hyperoperation.lean

```diff
@@ -0,0 +1,109 @@
+/-
+Copyright (c) 2023 Mark Andrew Gerads. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mark Andrew Gerads, Junyan Xu, Alex J. Best
+-/
+import tactic.ring
+
+/-!
+# Hyperoperation sequence
+
+This file defines the Hyperoperation sequence.
+`hyperoperation 0 m k = k + 1`
+`hyperoperation 1 m k = m + k`
+`hyperoperation 2 m k = m * k`
+`hyperoperation 3 m k = m ^ k`
+`hyperoperation (n + 1) m (k + 1) = hyperoperation n m (hyperoperation (n + 1) m k)`
+
+## References
+
+* <https://en.wikipedia.org/wiki/Hyperoperation>
+
+## Tags
+
+hyperoperation
+-/
+
+/--
+Implementation of the hyperoperation sequence
```

Comment:
This docstring should explain what `hyperoperation n m k` means with reference to each variable name.

Reply:
Is this docstring acceptable now? I am repeating what was said earlier, so I am unsure.

Reply:
What I mean is that you should say something like

> `hyperoperation n m k` is the `n`th hyperoperation between  `m` and `k`

Or whatever you would actually say in english.

Reply:
Now is fine?